docs/cartesian-product.adoc

```diff
@@ -18,26 +18,28 @@ Normally, this results in a lot of written test data parameters.
 For a more comfortable way you may use the `@CartesianProductTest` extension.
 The extension takes the test data parameter values and runs the test for every possible combination of them.
 
+NOTE: The CartesianProductTest extension has undergone significant changes in 1.4.3 (??? TODO: Fix).
+
 == Basic Use
 
 `@CartesianProductTest` is used _instead_ of `@Test` or other such annotations (e.g. `@RepeatedTest`).
 
 You can supply test parameters to `@CartesianProductTest` in multiple ways.
 
  - the annotation can have a `String[]` value (see <<Supplying CartesianProductTest with a `String[]`>>)
- - the test method can be annotated with `@CartesianValueSource`, `@CartesianEnumSource`, or range source annotations (see <<Annotating your test method>>)
+ - the test parameters can be annotated with `@CartesianValueSource`, `@CartesianEnumSource`, or range source annotations (see <<Annotating your test parameters>>)
  - the test class can have a static factory method providing the arguments (see <<Writing a static factory method for the parameters>>)
 
-Specifying more than one kind of parameter source (e.g.: both annotating your test method and having a static factory) does not work and will throw an `ExtensionConfigurationException`.
+Specifying more than one kind of parameter source (e.g.: both annotating your test parameters and having a static factory) does not work and will throw an `ExtensionConfigurationException`.
 
 Our earlier example with `{ 1, 2 }` and `{ 3, 4 }`, would look like this:
 
 [source,java]
 ----
 @CartesianProductTest
-@CartesianValueSource(ints = { 1, 2 })
-@CartesianValueSource(ints = { 3, 4 })
-void myCartesianTestMethod(int x, int y) {
+void myCartesianTestMethod(
+		@CartesianValueSource(ints = { 1, 2 }) int x,
+		@CartesianValueSource(ints = { 3, 4 }) int y) {
```

Comment:
`@Values` is too short in my opinion - it does not properly show that it belongs to the cartesian test extension. `@CartesianTest` would be fine, but that does not help at all with the actual issue.

I'm thinking to maybe re-use the parameterized test annotation names, e.g.: `@ValueSource`. Maybe `@CartesianValues`? But I'm iffy about `@Values` even if we move it to a dedicated package. I think it would lead to confusion.

Reply:
Re: moving `CartesianProductTest` to a separate package - I'm for it. It's larger than any other extension and it keeps on growing.

Reply:
> But I'm iffy about `@Values` even if we move it to a dedicated package. I think it would lead to confusion.

What about having `@Values` as inner annotation of `@CartesianProductTest`, similar to JUnit 4 [`@Parameters`](https://github.com/junit-team/junit4/blob/9ad61c6bf757be8d8968fd5977ab3ae15b0c5aba/src/main/java/org/junit/runners/Parameterized.java#L211-L233)?

Not my absolute preference, but with a static import might be acceptable.

Reply:
> What about having @Values as inner annotation of @CartesianProductTest, similar to JUnit 4 @Parameters?

> ##  similar to JUnit 4 @Parameters

> # JUnit 4

😔

Reply:
Once again, I came around on the idea, so it now works exactly like that. Thank you, @scordio!